### PR TITLE
feat(message-input): edit message mode using the same recipe 

### PR DESCRIPTION
**Pull request by sanjeevdialpad**
feat(message-input): Edit message mode using the same recipe 

## :hammer_and_wrench: Type Of Change

- [ ] Fix
- [x] Feature
- [ ] Refactoring
- [ ] Documentation

## :book: Description

Introduced new props to message input:
- isEdit : edit mode when true
- cancelButtonAriaLabel - aria label for the button
- cancelButtonText - Button text i18n translated

Events:
- cancel - when cancel button is pressed in edit mode

Screenshot 2023-11-15 at 3 18 07 PM

### CODE REVIEW

File: recipes/conversation_view/message_input/message_input.vue [hunk 1]
```diff
@@ -191,19 +213,30 @@
...
                   <dt-icon
                     name="send"
                     size="300"
                   />
                 </template>
+                <template
+                  v-if="isEdit"
+                >
+                  <p>{{ 'Save Changes' }}</p>
+                </template>
```
Comment (braddialpad) [inline recipes/conversation_view/message_input/message_input.vue]: Did we want to center middle the tooltip in this case now that it is wider?

Screenshot 2023-11-15 at 3 29 03 PM

**Review by braddialpad — APPROVED** [review]
Looks good, approving with one question.

File: recipes/conversation_view/message_input/message_input.vue [hunk 2]
```diff
@@ -191,19 +213,30 @@
...
+                >
                   <dt-icon
                     name="send"
                     size="300"
                   />
                 </template>
+                <template
+                  v-if="isEdit"
+                >
+                  <p>{{ 'Save Changes' }}</p>
```
Comment (braddialpad) [inline recipes/conversation_view/message_input/message_input.vue]: Oh yeah.. this needs to be i18n prop?

Comment (sanjeevdialpad) [reply recipes/conversation_view/message_input/message_input.vue]: thanks, i missed this.,

File: recipes/conversation_view/message_input/message_input.vue (hunk 1 above)
Comment (sanjeevdialpad) [reply recipes/conversation_view/message_input/message_input.vue]: when its edit mode, the tooltip will read save changes, so it still will look wider with that tooltip, hence kept the position same.

File: recipes/conversation_view/message_input/message_input.vue [hunk 3]
```diff
@@ -166,19 +167,40 @@
...
             :show="!isSendDisabled && sendButtonFocus"
-            :offset="[6, -4]"
+            :offset="[6, -8]"
           >
             <template #anchor>
               <!-- Right positioned UI - send button -->
               <dt-button
                 data-qa="dt-message-input-send-btn"
                 size="sm"
                 :kind="!isSendDisabled ? 'default' : 'muted'"
```
Comment (iropolo) [inline recipes/conversation_view/message_input/message_input.vue]: Will suggest to create a computed property here with the name of `buttonKind`

**Comment by iropolo** [comment]
Percy fails with: a new snapshot and some changes if they are intended, ok, there is a chance where they are flaky so its ok to go.

**Comment by braddialpad** [comment]
> Percy fails with: a new snapshot and some changes if they are intended, ok, there is a chance where they are flaky so its ok to go.

I think this will be fixed when he updates branch from staging-vue3

File: recipes/conversation_view/message_input/message_input.vue (hunk 3 above)
Comment (sanjeevdialpad) [reply recipes/conversation_view/message_input/message_input.vue]: done

**Pull request merged**